tecton_client/_internal/data_types.py

```diff
@@ -0,0 +1,60 @@
+from dataclasses import dataclass
+from typing import Dict
+from typing import List
+from typing import Optional
+
+
+@dataclass
+class GetFeaturesResult:
+    """Temp"""
+
+    features: list
+
+
+@dataclass
+class GetFeaturesResponse:
+    """Temp"""
+
+    result: GetFeaturesResult
+    metadata: Optional[Dict] = None
+
+    @classmethod
+    def from_dict(cls, resp: dict) -> "GetFeaturesResponse":
+        return GetFeaturesResponse(result=GetFeaturesResult(**resp["result"]), metadata=resp.get("metadata"))
+
+
+@dataclass
+class GetFeatureServiceMetadataResponse:
+    """Temp"""
+
+    feature_service_type: str
+    input_join_keys: List[dict]
+    input_request_context_keys: List[dict]
+    feature_values: List[dict]
+
+    @classmethod
+    def from_dict(cls, resp: dict):
+        """temp"""
+        return GetFeatureServiceMetadataResponse(
+            feature_service_type=resp["featureServiceType"],
+            input_join_keys=resp["inputJoinKeys"],
+            input_request_context_keys=resp["inputRequestContextKeys"],
+            feature_values=resp["featureValues"],
+        )
+
+
+class MetadataOptions:
```

Comment:
Looking at this, I'm concerned 
(1) Users will be confused at the usage of a class instead of an Enum. Regardless of their merits, purely in terms of self-documenting code I think a user would probably expect an Enum.
(2) Either class or enum, I think the constants should be capitalized. (This aligns with the starting example in the Python enum documentation.) 

Alternatively, and this aligns w/ a comment of Brian's below, the class could just be a container for options -- a glorified `dict`, where the user sets the non-defaults they would like. Then this class would own what the defaults are rather than inside the client. In this case neither of (1) or (2) above apply. Unfortunately I don't have a strong technical recommendation here that I can back up, so I'll leave it up to you.

Reply:
K, updated style. `get_features` no longer accepts a dictionary, it accepts an object. It's still a bit weird, but I think better than the enum thing, and less likely to cause programming issues.
```
 resp = client.get_features(
    feature_service_name="fraud_detection_feature_service:v2",
    join_key_map={"user_id": "user_4407104885"},
    request_context_map={"amount": 500.00},
    metadata_options=MetadataOptions(include_data_types=True, include_names=True),
)
```